packages/devextreme/js/__internal/ui/date_box/m_date_box.mask.parts.ts

```diff
@@ -79,6 +80,11 @@ const PATTERN_SETTERS = extend({}, getPatternSetters(), {
 
     date.setFullYear(newValue);
   },
+  x: (date, value) => {
+    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+    const adjustedMinutes = date.getMinutes() + value + date.getTimezoneOffset();
+    return new Date(date.setMinutes(adjustedMinutes));
```

Comment:
[Possibly excess code]
Why do we need to recreate the date, not just return parameter?

Reply:
[d0fea1b](https://github.com/DevExpress/DevExtreme/pull/27770/commits/d0fea1bf44e579b3abd09ba20a43d44e87904cc7)